app/population/middleware/set-population.js

```diff
@@ -44,6 +44,9 @@ async function setPopulation(req, res, next) {
       transfersOut,
     }
 
+    req.locationId = locationId
```

Comment:
When attaching the wizard to the router, all the params are lost, and the only one accessible is the `/edit` param.

These changes allow these variables to be used later on in the workflow.

Reply:
Have you tried using the [`mergeParams`](https://expressjs.com/en/api.html#express.router) option on the router? It might persist these through.

We've [used it elsewhere](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/3d5a283e30f8f22705ab76d23a32da8bace7519b/app/move/index.js#L4) to achieve something similar.

Reply:
Yeah, `mergeParams` is already being used on the subrouter, and it should work as its working on other parts of this route.

I've tried refactoring this a few times, changing the order of adding and applying and its broken every time :broken_heart:  

So given that it currently works, I don't really want to change this until I have E2E tests in place to protect me.

Reply:
Yeah, so I think it might actually need to be the [router created within the form wizard](https://github.com/HMPO/hmpo-form-wizard/blob/c8131187ddaafd220af89e75b8c2a400f4bdd78e/lib/wizard.js#L11) that needs that option, which might be why you can't access it.

Think this is good for now 👍